src/supabase/queries/cases.ts

```diff
@@ -37,6 +37,58 @@ export async function getCaseIdsFromUserId(
   }
 }
 
+export async function getCaseById(caseId: CaseUid): Promise<Case> {
+  try {
+    const { data } = await supabase.from('cases').select().eq('caseId', caseId);
+    if (!data) {
+      throw new Error('case not found');
+    }
+    return formatCaseProperly(data[0]);
+  } catch (error) {
+    console.warn('(getCaseById)', error);
+    throw error;
+  }
+}
+
+export async function isValidCase(caseId: CaseUid): Promise<boolean> {
+  const { data } = await supabase.from('cases').select().eq('caseId', caseId);
+  if (!data) {
+    return false;
+  }
+  return data.length !== 0;
+}
+
+export async function uploadCase(caseId: CaseUid): Promise<void> {
+  try {
+    const {
+      data: { user },
+    } = await supabase.auth.getUser();
+    const userId = user?.id;
+    await supabase.from('status').insert({ caseId, userId });
+  } catch (error) {
+    console.warn(error);
+    throw error;
+  }
+}
+
+export async function containsDuplicateCase(caseId: CaseUid): Promise<boolean> {
+  try {
```

Comment:
Do you think `try-catch` is necessary here? Not a big issue but `isValidCase` does not use a `try-catch` so let's be consistent and either do one or the other for these boolean cases!

Reply:
don't think so! I'll try it without the try-catch